### ISSUE TITLE
zdd: missing task IP in get_svnames_from_task

### ISSUE THREAD

**Issue by h0tbird** [mesosphere/marathon-lb#209]
A unique, friendly name for the backend server is forged [here](https://github.com/mesosphere/marathon-lb/blob/master/marathon_lb.py#L439-L452). In my case it looks like this `worker-1_cell-1_dub_xnood_com_10_0_0_243_31096`.

Later `zdd.py` forges `svnames` [here](https://github.com/mesosphere/marathon-lb/blob/master/zdd.py#L202-L205) with missing task IP so it looks like this `worker-1_cell-1_dub_xnood_com_31096`

Therefore [this](https://github.com/mesosphere/marathon-lb/blob/master/zdd.py#L228) condition is always false.

**Comment by JamesAwesome** [comment]
@brndnmtthws I think i see the problem here, could probably have a PR by tmw or friday

**Comment by h0tbird** [comment]
I guess the solution is to take the IP address from the `task` structure and concatenate it to the `prefix` variable defined in `get_svnames_from_task`. But that leads me to anticipate another problem in my setup.

This is a sample `task` structure:
```yaml
{
  "appId": "/nginx-blue",
  "healthCheckResults": [
    {
      "alive": true,
      "consecutiveFailures": 0,
      "firstSuccess": "2016-06-02T09:01:39.215Z",
      "lastFailure": null,
      "lastFailureCause": null,
      "lastSuccess": "2016-06-02T09:07:10.445Z",
      "taskId": "nginx-blue.9934795e-28a0-11e6-b2be-0242f110140f"
    }
  ],
  "host": "worker-1.cell-1.dub.xnood.com",
  "id": "nginx-blue.9934795e-28a0-11e6-b2be-0242f110140f",
  "ipAddresses": [
    {
      "ipAddress": "10.128.2.162",
      "protocol": "IPv4"
    }
  ],
  "ports": [
    31096
  ],
  "slaveId": "7a619ed4-e6e2-4fee-9d8b-ccfd2bb0fca2-S1",
  "stagedAt": "2016-06-02T09:01:31.180Z",
  "startedAt": "2016-06-02T09:01:33.903Z",
  "version": "2016-06-02T09:01:31.112Z"
}
```

The IP address reported is `10.128.2.162` which belongs to my overlay network (I use flannel at the moment) whereas the IP used by `marathon-lb.py` is `10.0.0.243` which is the Mesos agent IP, so it won't match neither.

I wish I could totally disable the inclusion of IP address in the backend names defined in HA-Proxy, does that make sense to you?

**Comment by h0tbird** [comment]
I have a second thought. Tell `marathon-lb.py` to use the container IP instead of the Mesos agent IP.

**Comment by h0tbird** [comment]
Your patch is an improvement and I think it is fare to close the issue.

But I still have the problem where:
`marathon-lb.py` produces `worker-2_cell-1_dub_xnood_com_10_0_0_124_31195` (agent IP)
`zdd.py` produces `worker-2_cell-1_dub_xnood_com_10_128_4_98_31195` (container IP)

I have _IP-per-container_ via flannel and not _IP-per-task_ as described [here](https://mesosphere.github.io/marathon/docs/ip-per-task.html).
I think there is not enough context at the time `get_task_ip_and_ports` is executed by `marathon-lb.py`. Perhaps flannel takes some time to assign the IP to the container, I don't know...

Any ideas would be greatly appreciated.

**Comment by h0tbird** [comment]
Problem solved by #218
